azure-mgmt-batch/src/main/java/com/microsoft/azure/management/batch/BatchAccount.java

```diff
@@ -119,21 +129,29 @@
         }
 
         /**
-         * A batch account definition with sufficient inputs to create a new
-         * batch account in the cloud, but exposing additional optional inputs to
-         * specify.
+         * A batch account definition to allow creation of application.
          */
-        interface WithCreate extends
-            Creatable<BatchAccount>,
-            Resource.DefinitionWithTags<WithCreate> {
+        interface WithApplication {
+            /**
+             * First stage to create new application in Batch account.
+             *
+             * @param applicationId id of the application to create
+             * @return next stage to create the Batch account.
+             */
+            Application.DefinitionStages.Blank<WithStorage> defineNewApplication(String applicationId);
+        }
 
+        /**
+         * A batch account definition to allow attaching storage accounts.
+         */
+        interface WithStorage {
             /**
              * Specifies that an existing storage account to be attached with the batch account.
              *
              * @param storageAccount existing storage account to be used
              * @return the stage representing creatable batch account definition
              */
-            DefinitionStages.WithCreate withStorageAccount(StorageAccount storageAccount);
+            DefinitionStages.WithCreate withExistingStorageAccount(StorageAccount storageAccount);
 
```

Comment:
Discussion: we may move storage account to optionals since all applications required to be hosted on single storage account.